upstairs/src/lib.rs

```diff
@@ -893,7 +858,47 @@ where
                             ).await;
                             bail!("expected version 1, got {}", version);
                         }
+
                         negotiated = 1;
+
+                        up.ds_repair_address(up_coms.client_id, repair_addr).await;
+
+                        // If this downstairs is returning, then attempt repair
+                        {
```

Comment:
I'm curious why you moved this inside here and away from the top of the function.  I don't know what the impact of moving the `ds.re_new()` call to this phase of negotiation will have, and by moving it down we require getting to this phase of negotiation before we re_new the work on the work queue.  There is also some coming work that will handle that panic case (the return of a failed downstairs).

If it's just because we want to print the repair address and we don't have the repair address yet, then we don't have to print it.  I was going to remove that anyway, but if we do want to print, we can just add a message inside this match to print it.

Also, nit:  > 80 characters.  Moving it back would fix that :)

Reply:
I was under the impression that `ds.re_new()` performed some part of the repair but that's wrong. Undid the move in 0c041ea